File: go/src/dataprocessor/task.go
package main // Important: Belongs to the main package now

import (
    "fmt"
    "math/rand"
    "time"
)

type Task struct {
    ID   string
    Data string
}

func (t *Task) Process() string {
    // Simulate processing time
    time.Sleep(time.Duration(rand.Intn(500)) * time.Millisecond)
    return fmt.Sprintf("Processed: %s (Task ID: %s)", t.Data, t.ID)
}

File: java/src/main/java/com/example/Task.java
package com.example;

import java.util.Random;

public class Task {
    private String id;
    private String data;

    public Task(String id, String data) {
        this.id = id;
        this.data = data;
    }

    public String getId() {
        return id;
    }

    public String getData() {
        return data;
    }

    public String process() {
        // Simulate processing time
        try {
            Thread.sleep(new Random().nextInt(500));
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
        return "Processed: " + data + " (Task ID: " + id + ")";
    }
}